File: 28.实现-str-str.py
#
# @lc app=leetcode.cn id=28 lang=python
#
# [28] 实现 strStr()
#

# @lc code=start
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        i = 0
        if len(needle)==0:
            return 0
        while i<= len(haystack) - len(needle):
            if haystack[i:i+len(needle)]==needle:
                return i
            i = i+1
        return -1
            
# @lc code=end

File: 28.实现-str-str.java
/*
 * @lc app=leetcode.cn id=28 lang=java
 *
 * [28] 实现 strStr()
 */

// @lc code=start
class Solution {
    public int strStr(String haystack, String needle) {
        int lenHay = haystack.length();
        int lenNee = needle.length();

        if(lenHay<lenNee){
            return -1;
        } else if(lenNee==0){
            return 0;
        }

        char[] arrHaystack = haystack.toCharArray();
        char[] arrNeedle = needle.toCharArray();
        char firstNeedle = arrNeedle[0];

        for(int i=0; i< lenHay-lenNee+1; i++){
            if(arrHaystack[i]==firstNeedle){
                String subHay = haystack.substring(i,i+lenNee);
                if(subHay.equals(needle)){
                    return i;
                }
            }
        }
        return -1;

        // return haystack.indexOf(needle);
    }
}
// @lc code=end

